index.js

```diff
@@ -41,22 +45,29 @@ export default class StarReview extends Component {
                 </View>
                 <Image style={{height: this.props.starSize, width: this.props.starSize, backgroundColor:'transparent', position:'absolute'}} source={this.props.emptyStar}/>
               </ImageBackground>
-            </View>
-          );
+            </View> ;
+
+          stars.push(halfStarComponent);
         }else if(i > Math.floor(this.props.value) + 1){
           //empty stars
-          stars.push(
+          let emptyStarComponent = this.isReactElement(this.props.emptyStar) ?
+            <View key={i}>{this.props.emptyStar}</View>
+            :
             <View key={i} style={{paddingLeft: this.props.spacing/2, paddingRight: this.props.spacing/2}}>
               <Image style={starStyle} source={this.props.emptyStar}/>
-            </View>
-          );
+            </View> ;
+
+          stars.push(emptyStarComponent);
         }else{
           //filled stars
-          stars.push(
+          let starComponent = this.isReactElement(this.props.fullStar) ?
+            <View key={i}>{this.props.fullStar}</View>
+            :
```

Comment:
Would you mind changing this to an if statement instead of a ternary? This conditional significantly changes the component's behavior so I'd like it to be obvious :)